src/core/org/apache/jmeter/gui/action/ActionRouter.java

```diff
@@ -67,6 +72,9 @@ public void actionPerformed(final ActionEvent e) {
 
     private void performAction(final ActionEvent e) {
         String actionCommand = e.getActionCommand();
+        if(!NO_TRANSACTION_ACTIONS.contains(actionCommand)) {
```

Comment:
Perhaps a small helper method `needsTransaction(actionCommand)` would help readability, as it would get rid of the double negation (`!NO..`)